### ISSUE TITLE
Entry point identifiers with single quote break C compilation

### ISSUE THREAD

**Issue by nbos** [diku-dk/futhark#1841]
Compiling for instance
```
entry f (x: f32) : f32 = x
entry f' (x: f32) : f32 = x+1
```
with a C generating backend like `multicore` raises
```
cc failed with code 1:
test.c:1989:30: warning: missing terminating ' character
 1989 | static int futrts_cli_entry_f'(struct futhark_context *ctx)
      |                              ^
test.c:1989:30: error: missing terminating ' character
 1989 | static int futrts_cli_entry_f'(struct futhark_context *ctx)
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test.c:1990:1: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘{’ token
 1990 | {
      | ^
test.c: In function ‘main’:
test.c:2122:126: warning: missing terminating ' character
 2122 |     struct entry_point_entry entry_points[] = {{.name ="f", .fun =futrts_cli_entry_f}, {.name ="f'", .fun =futrts_cli_entry_f'}};
      |                                                                                                                              ^
test.c:2122:126: error: missing terminating ' character
 2122 |     struct entry_point_entry entry_points[] = {{.name ="f", .fun =futrts_cli_entry_f}, {.name ="f'", .fun =futrts_cli_entry_f'}};
      |                                                                                                                              ^~~~
test.c:2124:5: error: expected ‘}’ before ‘if’
 2124 |     if (entry_point != NULL) {
      |     ^~
test.c:2122:88: note: to match this ‘{’
 2122 |     struct entry_point_entry entry_points[] = {{.name ="f", .fun =futrts_cli_entry_f}, {.name ="f'", .fun =futrts_cli_entry_f'}};
      |                                                                                        ^
In file included from test.c:2165:
/usr/include/assert.h:67:1: error: expected ‘}’ before ‘extern’
   67 | extern void __assert_fail (const char *__assertion, const char *__file,
      | ^~~~~~
test.c:2122:47: note: to match this ‘{’
 2122 |     struct entry_point_entry entry_points[] = {{.name ="f", .fun =futrts_cli_entry_f}, {.name ="f'", .fun =futrts_cli_entry_f'}};
      |                                               ^
test.c:7748:1: error: expected declaration or statement at end of input
 7748 | }
      | ^
```

**Comment by athas** [comment]
I could have sworn we used to have a warning or error about this.

**Comment by athas** [comment]
Fixed in 7cb7e64f66bc8923b2c437b123389d316cdc9f04.
